components/PageNavigation.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <b-navbar
+    class="navigation-container border-bottom"
+    data-qa="main navigation"
+  >
+    <b-container class="p-0">
+      <b-navbar-nav class="px-2">
+        <li
+          v-for="nav in navigation"
+          :key="nav.text"
+          class="nav-item"
+        >
+          <SmartLink
+            :destination="localePath({name: 'slug', params: { pathMatch: nav.url.replace(/^\//, '') }})"
+            link-class="nav-link"
+            exact-active-class="font-weight-bold"
+          >
+            {{ nav.text }}
+          </SmartLink>
+        </li>
+      </b-navbar-nav>
+    </b-container>
+  </b-navbar>
+</template>
+
+<script>
+  import SmartLink from './generic/SmartLink';
+
+  export default {
+    components: {
+      SmartLink
+    },
+
+    computed: {
+      navigation() {
+        return this.$store.state.navigation.data;
+      },
+
+      i18n() {
+        return this.$store.state.i18n.locale;
+      }
+    },
+
+    watch: {
+      i18n() {
+        this.getNavigationData();
+      }
+    },
+
+    mounted() {
+      this.getNavigationData();
+    },
+
+    methods: {
+      async getNavigationData() {
+        return this.$store.dispatch('navigation/init');
```

Comment:
add this, or rather `    await dispatch('navigation/init');` to store/index.js the same as the footer, that way it avoids the loading in only happening after the page already rendered. Because now it jumps the whole page down on a hard referesh. We'll still need this getNavigationData as otherwise it won't pick up on language switchting.

Reply:
The reason for putting this in a method was due to language selector. If we use `nuxtServerInit` method, the language doesn't update when the user changes the language. Hope that makes sense.

Reply:
yes, as I was saying we'll want it in both places.

Reply:
we could attempt to skip the mounted call when the local hasn't changed, but I think this is probably overkill at this point.

Reply:
Apologies.. misread

Reply:
Dont think mounted is actually need if we are going to use NuxtServerInit